### PR TITLE
http request, json parsing and displaying

### PR DESCRIPTION
**Pull request by tlouwers**
can send a http request to jenkins and receive a job status in json format. this json can then be parsed and the leds can be turned on accordingly.

### CODE REVIEW

File: buildserver-monitor/Application.cpp [hunk 1]
```diff
@@ -31,8 +31,12 @@
 Application::Application() :
     mLeds(mLogger),
     mSM(mLogger),
-    mWifi(mLogger)
-{ ; }
+    mWifi(mLogger),
+    mHttp(mLogger),
+    mState(BuildState::NoState)
```
Comment (tlouwers) [inline buildserver-monitor/Application.cpp]: For clarity: mBuildState instead of mState.

File: buildserver-monitor/Application.cpp [hunk 2]
```diff
@@ -228,9 +233,13 @@ bool Application::TryAcquiring()
 
     mLeds.SetColor(3, LedColor::Blue);
     delay(1000);
+
+    bool result = mHttp.Acquire();
+    Serial.println("hello there");
```
Comment (tlouwers) [inline buildserver-monitor/Application.cpp]: ?

File: buildserver-monitor/httpClient.cpp [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  mLogger(logger)
+{
+  mInit = false;
+  mJsonString = "";
+}
+
+/**
+   \brief   Destructor, disconnects (if connected).
+*/
+httpClient::~httpClient()
```
Comment (tlouwers) [inline buildserver-monitor/httpClient.cpp:51]: Can be kept in the header if not doing anything here.

File: buildserver-monitor/httpClient.cpp [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+{
+
+}
+
+bool httpClient::Init()
+{
+  if (!mInit) {
+    mLogger.Log(LogLevel::INFO, "Initializing http client");
+    client = new BearSSL::WiFiClientSecure();
+    client->setFingerprint(SHA1_FINGERPRINT);
```
Comment (tlouwers) [inline buildserver-monitor/httpClient.cpp:65]: Add check for nullptr, maybe device is out of resources. A log entry when that happens and failure result to enter the error state would be nice as well.

File: buildserver-monitor/httpClient.hpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+  private:
+    ILogging& mLogger;
+
+    HTTPClient https;
+    BearSSL::WiFiClientSecure* client;
+
+    bool mInit;
+    String mJsonString;
+    char* mResult;
```
Comment (tlouwers) [inline buildserver-monitor/httpClient.hpp:75]: I think you mean: "const char* mResult", or maybe even: "const char* mBuildResult"

File: buildserver-monitor/http_config.h [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+#endif
+
+/************************************************************************/
+/* Includes                                                             */
+/************************************************************************/
+#include <stdint.h>
+#include <stddef.h>
+
+// Network
+static std::string    USERNAME = "your username";     // The jenkins username.
```
Comment (tlouwers) [inline buildserver-monitor/http_config.h]: For robustness, a check in the httpClient that the default values presented here are not used (but properly filled) would be handy.

File: buildserver-monitor/httpClient.hpp (hunk 5 above)
Comment (jipsate) [reply buildserver-monitor/httpClient.hpp:75]: If i were to make this const I wouldn't be able to change the content of the string. Which is what i want to be able to do.

**Comment by tlouwers** [comment]
Apparently I created the pull request, will merge the rework.

**Pull request merged**